lib/iso_lang.ex

```diff
@@ -118,4 +116,111 @@ defmodule IsoLang do
       {:error, error} -> raise error
     end
   end
+
+  @doc """
+  Checks if query is valid against the standard provided. This query is case-sensitive.
+  If the `:by` field is not specified, the default will be used.
+
+  ## Options
+
+  - `:by` specifies which struct field to be used in the search. Default: `:alpha2`
+
+  ## Examples
+
+      iex> IsoLang.valid?("en")
+      true
+
+      iex> IsoLang.valid?("en", by: :alpha3b)
+      false
+
+      iex> IsoLang.valid?("eng", by: :alpha3b)
+      true
+  """
+
+  @spec valid?(query :: String.t(), opts :: Keyword.t()) :: true | false
+  def valid?(query, opts \\ []) do
+    opts =
+      opts
+      |> Keyword.merge(by: Keyword.get(opts, :by, :alpha2))
+
+    query
+    |> get(opts)
+    |> case do
+      {:ok, _} -> true
+      _ -> false
+    end
+  end
+
+  @doc """
+  Converts from name to provided ISO-639 standard
```

Comment:
This is too confusing a description.  It's not really converting so much as it's looking up a 2-letter code using the full language name.  But how/why is this checking the other fields?  Why does `code("en")` return an error if it is checking the `:alpha2` field?